staging/src/k8s.io/apimachinery/pkg/util/validation/field/errors.go

```diff
@@ -123,6 +123,18 @@ const (
 	// ErrorTypeInternal is used to report other errors that are not related
 	// to user input.  See InternalError().
 	ErrorTypeInternal ErrorType = "InternalError"
+	// TooLongFailCode is for MaxLengthExceed error
+	// This error is CEL blocking errors. The CEL validation will be skipped when this error occurred.
+	TooLongFailCode ErrorType = "MaxLengthExceed"
+	// MaxItemsFailCode is for MaxItemsExceed error
+	// This error is CEL blocking errors. The CEL validation will be skipped when this error occurred.
+	MaxItemsFailCode ErrorType = "MaxItemsExceed"
+	// TooManyPropertiesCode is for MaxPropertiesExceed error
+	// This error is CEL blocking errors. The CEL validation will be skipped when this error occurred.
+	TooManyPropertiesCode ErrorType = "MaxPropertiesExceed"
```

Comment:
could these use the existing ErrorTypeTooLong instead?

Reply:
I was trying to not change the previous behavior. The previous behavior, those errors will be returned back to user with `invalid value` indicate at beginning. I guess to indicate `too long` instead of `invalid value` makes sense for those errs. Will update.

Reply:
After revisit, TooLong/TooMany might not be the best for representing maxLengthExceed/maxItemsExceed/maxPropertiesExceed err due to 1. It changed the current behavior of returning `invalid value` to indicate the err to `Too long/Too many`. 2. It requires update to current way of generating the error. I have updated to have TooLongCode/TooManyCode as the error types and please feel free to let me know if returning `too long/too many` instead of `invalid value` is strong preferred.

Reply:
I still prefer using the TooMany / TooLong codes... if we need alternate constructors that tolerate unknown actual/max values and construct messages slightly differently, that's fine.

As a follow-up, we should update kube-openapi to plumb the info we need here so we can have good error messages. Example in https://github.com/liggitt/kubernetes/commits/placeholder

Reply:
I have added TODO for kube-openapi err handling. And I have updated to use error type `ErrorTypeTooLong/ErrorTypeTooMany` for existing err